src/components/scenes/SendScene2.tsx

```diff
@@ -0,0 +1,924 @@
+import {
+  asMaybeInsufficientFundsError,
+  asMaybeNoAmountSpecifiedError,
+  EdgeAccount,
+  EdgeCurrencyWallet,
+  EdgeDenomination,
+  EdgeSpendInfo,
+  EdgeSpendTarget,
+  EdgeTransaction
+} from 'edge-core-js'
+import * as React from 'react'
+import { Alert, TextInput, View } from 'react-native'
+import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view'
+import { sprintf } from 'sprintf-js'
+
+import { selectWalletForExchange } from '../../actions/CryptoExchangeActions'
+import { dismissScamWarning } from '../../actions/ScamWarningActions'
+import { playSendSound } from '../../actions/SoundActions'
+import { FIO_STR, getSpecialCurrencyInfo } from '../../constants/WalletAndCurrencyConstants'
+import { useAsyncEffect } from '../../hooks/useAsyncEffect'
+import { useDisplayDenom } from '../../hooks/useDisplayDenom'
+import { useExchangeDenom } from '../../hooks/useExchangeDenom'
+import { useHandler } from '../../hooks/useHandler'
+import { useWatch } from '../../hooks/useWatch'
+import s from '../../locales/strings'
+import { addToFioAddressCache, checkRecordSendFee, FIO_NO_BUNDLED_ERR_CODE, recordSend } from '../../modules/FioAddress/util'
+import { useState } from '../../types/reactHooks'
+import { useDispatch, useSelector } from '../../types/reactRedux'
+import { Actions, NavigationProp, RouteProp } from '../../types/routerTypes'
+import { GuiExchangeRates } from '../../types/types'
+import { getCurrencyCode, getTokenId } from '../../util/CurrencyInfoHelpers'
+import { getWalletName } from '../../util/CurrencyWalletHelpers'
+import { logActivity } from '../../util/logger'
+import { convertTransactionFeeToDisplayFee, DECIMAL_PRECISION, roundedFee } from '../../util/utils'
+import { WarningCard } from '../cards/WarningCard'
+import { SceneWrapper } from '../common/SceneWrapper'
+import { ButtonsModal } from '../modals/ButtonsModal'
+import { FlipInputModal2, FlipInputModalRef, FlipInputModalResult } from '../modals/FlipInputModal2'
+import { TextInputModal } from '../modals/TextInputModal'
+import { WalletListModal, WalletListResult } from '../modals/WalletListModal'
+import { Airship, showError } from '../services/AirshipInstance'
+import { cacheStyles, Theme, useTheme } from '../services/ThemeContext'
+import { EdgeText } from '../themed/EdgeText'
+import { ExchangedFlipInputAmounts, ExchangeFlipInputFields } from '../themed/ExchangedFlipInput2'
+import { PinDots } from '../themed/PinDots'
+import { SafeSlider } from '../themed/SafeSlider'
+import { SelectFioAddress2 } from '../themed/SelectFioAddress2'
+import { AddressTile2, ChangeAddressResult } from '../tiles/AddressTile2'
+import { EditableAmountTile } from '../tiles/EditableAmountTile'
+import { ErrorTile } from '../tiles/ErrorTile'
+import { Tile } from '../tiles/Tile'
+
+interface Props {
+  navigation: NavigationProp<'send2'>
+  route: RouteProp<'send2'>
+}
+
+export interface SendScene2Params {
+  walletId: string
```

Comment:
Not sure I can do this since the scene allows users to change the wallet while still in the scene. withWallet won't auto update the wallet in such a case.